sdk/metricsadvisor/azure-ai-metricsadvisor/azure/ai/metricsadvisor/_metrics_advisor_administration_client.py

```diff
@@ -1231,173 +1236,176 @@ def list_data_feed_ingestion_status(
+            ~azure.ai.metricsadvisor.models.DatasourceServicePrincipalInKeyVault]
         :raises ~azure.core.exceptions.HttpResponseError:
 
         .. admonition:: Example:
 
-            .. literalinclude:: ../samples/sample_credential_entities.py
-                :start-after: [START get_credential_entity]
-                :end-before: [END get_credential_entity]
+            .. literalinclude:: ../samples/sample_datasource_credentials.py
+                :start-after: [START get_datasource_credential]
+                :end-before: [END get_datasource_credential]
                 :language: python
                 :dedent: 4
-                :caption: Get a credential entity by its ID
+                :caption: Get a datasource credential by its ID
         """
 
-        credential_entity = self._client.get_credential(credential_entity_id, **kwargs)
-        return convert_to_credential_entity(credential_entity)
+        datasource_credential = self._client.get_credential(credential_id, **kwargs)
+        return convert_to_datasource_credential(datasource_credential)
 
     @distributed_trace
-    def create_credential_entity(
-            self, credential_entity,    # type: CredentialEntityUnion
+    def create_datasource_credential(
+            self, datasource_credential,    # type: DatasourceCredentialUnion
             **kwargs  # type: Any
     ):
-        # type: (...) -> CredentialEntityUnion
-        """Create a new data source credential entity.
-
-        :param credential_entity: The data source credential entity to create
-        :type credential_entity: Union[~azure.ai.metricsadvisor.models.SqlConnectionStringCredentialEntity,
-            ~azure.ai.metricsadvisor.models.DataLakeGen2SharedKeyCredentialEntity,
-            ~azure.ai.metricsadvisor.models.ServicePrincipalCredentialEntity,
-            ~azure.ai.metricsadvisor.models.ServicePrincipalInKeyVaultCredentialEntity]
-        :return: The created data source credential entity
-        :rtype: Union[~azure.ai.metricsadvisor.models.SqlConnectionStringCredentialEntity,
-            ~azure.ai.metricsadvisor.models.DataLakeGen2SharedKeyCredentialEntity,
-            ~azure.ai.metricsadvisor.models.ServicePrincipalCredentialEntity,
-            ~azure.ai.metricsadvisor.models.ServicePrincipalInKeyVaultCredentialEntity]
+        # type: (...) -> DatasourceCredentialUnion
+        """Create a new datasource credential.
+
+        :param datasource_credential: The datasource credential to create
+        :type datasource_credential: Union[~azure.ai.metricsadvisor.models.DatasourceSqlConnectionString,
+            ~azure.ai.metricsadvisor.models.DatasourceDataLakeGen2SharedKey,
+            ~azure.ai.metricsadvisor.models.DatasourceServicePrincipal,
+            ~azure.ai.metricsadvisor.models.DatasourceServicePrincipalInKeyVault]
+        :return: The created datasource credential
+        :rtype: Union[~azure.ai.metricsadvisor.models.DatasourceSqlConnectionString,
```

Comment:
should the base type DataSourceCredential be included in the return type here?

Reply:
Johan mentioned Datasource here is one word. :)

Reply:
sorry, should `DatasourceCredential` be added to the rtype Union?

Reply:
This is the feedback from arch board review. Given those confusing and growing string for id, we want user to use something like delete(str) and not care about what is the exact arg name xxx_id. Also this can help us prevent breaking changes if we want to change the arg name.

Reply:
For create method, I don't think so. It should return the same type as the one from input.